.github/actions/run-interop-ping-test/action.yml

```diff
@@ -9,9 +9,38 @@ inputs:
     description: "Space-separated paths to JSON files describing additional images"
     required: false
     default: ""
+  s3-cache-bucket:
+    description: "Which S3 bucket to use for container layer caching"
+    required: false
+    default: ""
+  s3-access-key-id:
+    description: "S3 Access key id for the cache"
+    required: false
+    default: ""
+  s3-secret-access-key:
+    description: "S3 secret key id for the cache"
+    required: false
+    default: ""
+  aws-region:
+    description: "Which AWS region to use"
+    required: false
+    default: "us-east-1"
 runs:
   using: "composite"
   steps:
+    - run: |
+        echo "AWS_BUCKET=${{ inputs.s3-cache-bucket }}" >> $GITHUB_ENV
+        echo "AWS_REGION=${{ inputs.aws-region }}" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Configure AWS credentials for S3 build cache
+      if: ${{ inputs.s3-access-key-id }} != "" && ${{ inputs.s3-secret-access-key }} != ""
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ inputs.s3-access-key-id }}
+        aws-secret-access-key: ${{ inputs.s3-secret-access-key}}
+        aws-region: ${{ env.AWS_REGION }}
```

Comment:
rust-libp2p executes this step on each pull request. Some pull requests originate from forks. Those pull requests don't have access to the above secrets. Thus this step fails.

See https://github.com/libp2p/rust-libp2p/actions/runs/4142906998/jobs/7164133938 on https://github.com/libp2p/rust-libp2p/pull/3344 as an example.

I don't yet have the full picture on the caching approach via an S3 bucket. Would mounting the bucket as read-only from forks be an option? Worst case, we don't mount the bucket on forks at all, thus building all images from scratch.

Reply:
> Thus this step fails.

The original plan was that the `if` should have skipped this step. I'll check what that isn't happening.

> Would mounting the bucket as read-only from forks be an option? 

Unfortunately no. The s3 cache needs rw access. Maybe there's a hidden option I'm not seeing, but when I tried a read only key it failed.

My original plan was to have forks build from scratch.

Reply:
I had the `if` statement wrong. #131 fixes it.

Reply:
Did you try omitting the `--cache-to` option when you tried with a read-only key?

According to https://docs.docker.com/build/cache/backends/, those two options operate separately and I am surprised that `--cache-from` by itself would need write permissions.

If we conditionally define `--cache-to`, we should be set our bucket to be publicly readable and have forks use our cache.

Alternatively, I see that this also has a registry cache option which might be interesting to explore.

I really want our CI to always be fast, not just for our own PRs. (I also don't like wasting compute power.)

Reply:
Happy to help with this but it would earliest be in ~5 days as I am off the grid for the next few.

Reply:
> Did you try omitting the --cache-to option when you tried with a read-only key?

I hadn't, that's probably it!

Probably worth figuring out how to make this bucket public and then defaulting to that public bucket if no keys are passed. I'll open an issue.